feat: rewrite ujust harden-flatpak in Python

HastD opened the pull request:
The Python rewrite has the same basic logic but with some improvements to user-facing messages and the app ID search logic.

The third-party `inquirer` module is used for letting the user select from a list of options, and is installed via RPM. (I put it in `common-packages.yml` rather than `desktop-packages.yml` because it's likely to be useful in other ujust scripts, and is a small package in any case.)

Also move audit-script-specific parts of the `utils` module into a new `audit_utils` module, and fix a minor type error in `is_using_vpn`.

HastD committed:
feat: rewrite ujust harden-flatpak in Python

The Python rewrite has the same basic logic but with some improvements
to user-facing messages and the app ID search logic.

The third-party `inquirer` module is used for letting the user select
from a list of options, and is installed via RPM. (I put it in
`common-packages.yml` rather than `desktop-packages.yml` because it's
likely to be useful in other ujust scripts, and is a small package in
any case.)

Also move audit-script-specific parts of the `utils` module into a new
`audit_utils` module, and fix a minor type error in `is_using_vpn`.

RoyalOughtness commented on files/justfiles/desktop/flatpak.just:
```diff
@@ -23,76 +23,10 @@ disable-flathub-unfiltered:
-
-    harden_flatpak_app "$flatpak_id"
-    echo "$hmalloc_description applied to flatpak $flatpak_id"
+[positional-arguments]
```
is this tag supported? https://github.com/casey/just/issues/1651#issuecomment-1663091583

HastD replied:
I think CI is installing an older version of `just` or something; I tested it locally and it works fine. However, just to be on the safe side and avoid errors in CI, I'll switch to using the other method. (None of the other recipes in `flatpak.just` use command-line arguments, so it won't affect them to apply `positional-arguments` for the whole justfile.)

RoyalOughtness replied:
ah, yeah ubuntu probably just has an old version

RoyalOughtness commented:
Aside from the cyclomatic complexity codacy complaints (which are almost certainly already present in the bash), this seems good

HastD commented:
The cyclomatic complexity issues being flagged are due to the difference between Lizard (the complexity analyzer Codacy uses) and the one Ruff uses; this issue has come up before, and I still think the one Ruff uses does a better job of measuring the complexity of Python code. The main difference I've noticed is that with Lizard, list comprehensions increase cyclomatic complexity as if they were written as loops.

Ruff is set up to check that all functions have cyclomatic complexity at most 8, and the code here passes that check.

HastD committed:
fix: use [positional-arguments] tag instead of global setting

HastD committed:
chore: use manuallyinstalljust.sh in tests

This ensures the version of `just` used in CI is the same as the one
used in the images. Also made some minor reliability improvements to
`manuallyinstalljust.sh`: use `mktemp` to generate a temporary directory
instead of just using `/tmp`, and set `curl` options to retry on failure
and to not suppress error message output.

ShadowSlayer1441 commented on files/system/desktop/usr/libexec/secureblue/harden_flatpak.py:
```diff
@@ -0,0 +1,186 @@
+        print("No matching app IDs found; exiting.")
+        return 1
+    harden_flatpak_app(app_id, hmalloc_path)
+    print(f"{hmalloc_description} applied to flatpak {app_id}")
```
This should mention the filesystem perm change

HastD replied:
Is that worth mentioning? `filesystem=host-os:ro` just grants read-only access to `/usr` (and `/etc/ld.so.cache` and `/etc/alternatives`), so it's a very benign permission and I think of it as essentially an implementation detail of what it means to "apply hardened_malloc" to a flatpak. (Similarly, the message doesn't mention setting the environment variable `LD_PRELOAD`, even though that's how it's done.)

ShadowSlayer1441 replied:
Maybe not, but if we specifically mention hmalloc being applied, then we imply that no other changes were made which is not accurate. I would say it's more about informing the user than host-os:ro  mattering.

HastD replied:
Makes sense, I'll add a message about that.

ShadowSlayer1441 commented on files/system/usr/libexec/secureblue/audit_utils/__init__.py:
```diff
@@ -0,0 +1,184 @@
+    await sub.wait()
+    # pylint: disable=use-implicit-booleaness-not-comparison-to-zero
+    if check and sub.returncode != 0:
+        err = f"async command `{cmd} {' '.join(args)}` returned nonzero exit code {sub.returncode}"
```
Why not print the stderr output to help the user fix the auditing issue? Esp. since this is probably some kind of permissions issue the user needs to fix versus a script bug.

HastD replied:
This PR only moves this function to a separate module (without changing it) for organizational purposes. Given that this function is only used in the audit script, not in `ujust harden-flatpak`, I think that change would be out of scope for this PR.

HastD committed:
chore: text wrapping, add note about host-os:ro permission
